adapters/repos/db/shard_geo_props.go

```diff
@@ -93,3 +93,73 @@ func (s *Shard) makeCoordinatesForID(propName string) geo.CoordinatesForID {
 func geoPropID(shardID string, propName string) string {
 	return fmt.Sprintf("%s_%s", shardID, propName)
 }
+
+func (s *Shard) updatePropertySpecificIndices(object *storobj.Object,
+	status objectInsertStatus) error {
+	for propName, propIndex := range s.propertyIndices {
+		if err := s.updatePropertySpecificIndex(propName, propIndex,
+			object, status); err != nil {
+			return errors.Wrapf(err, "property %q", propName)
+		}
+	}
+
+	return nil
+}
+
+func (s *Shard) updatePropertySpecificIndex(propName string,
+	index propertyspecific.Index, obj *storobj.Object,
+	status objectInsertStatus) error {
+	if index.Type != schema.DataTypeGeoCoordinates {
+		return fmt.Errorf("unsupported per-property index type %q", index.Type)
+	}
+
+	// currently the only property-specific index we support
+	return s.updateGeoIndex(propName, index, obj, status)
+}
+
+func (s *Shard) updateGeoIndex(propName string, index propertyspecific.Index,
+	obj *storobj.Object, status objectInsertStatus) error {
+	if status.docIDChanged {
+		if err := s.deleteFromGeoIndex(index, status.oldDocID); err != nil {
+			return errors.Wrap(err, "delete old doc id from geo index")
+		}
+	}
+
+	return s.addToGeoIndex(propName, index, obj, status)
+}
+
+func (s *Shard) addToGeoIndex(propName string, index propertyspecific.Index,
+	obj *storobj.Object, status objectInsertStatus) error {
+	if obj.Schema() == nil {
+		return nil
+	}
+
+	asMap := obj.Schema().(map[string]interface{})
+	propValue, ok := asMap[propName]
+	if !ok {
+		return nil
+	}
+
+	// geo coordinates is the only supported one at the moment
+	asGeo, ok := propValue.(*models.GeoCoordinates)
+	if !ok {
+		return fmt.Errorf("expected prop to be of type %T, but got: %T",
+			&models.GeoCoordinates{}, propValue)
+	}
+
+	if err := index.GeoIndex.Add(int(status.docID), asGeo); err != nil {
+		return errors.Wrapf(err, "insert into geo index")
+	}
+
+	return nil
+}
+
+func (s *Shard) deleteFromGeoIndex(index propertyspecific.Index,
+	docID uint32) error {
+	fmt.Printf("delete doc id %d\n", docID)
```

Comment:
I think this line should be also deleted

Reply:
:D Probably a leftover from furious `Prinft` debugging -)